src/System.Diagnostics.DiagnosticSource/src/System/Diagnostics/HttpHandlerDiagnosticListener.cs

```diff
@@ -557,33 +558,69 @@ private HttpHandlerDiagnosticListener() : base(DiagnosticListenerName)
 
         private void RaiseRequestEvent(HttpWebRequest request)
         {
-            // If System.Net.Http.Request is on, raise the event
-            if (this.IsEnabled(RequestWriteName))
+            if (request.Headers.Get(RequestIdHeaderName) != null)
             {
-                long timestamp = Stopwatch.GetTimestamp();
-                this.Write(RequestWriteName,
-                    new
+                // this request was instrumented by previous RaiseRequestEvent
+                return;
+            }
+
+            if (this.IsEnabled(ActivityName, request))
+            {
+                var activity = new Activity(ActivityName);
+
+                // Only send start event to users who subscribed for it, but start activity anyway
+                if (this.IsEnabled(RequestStartName))
+                {
+                    long timestamp = Stopwatch.GetTimestamp();
```

Comment:
I would like to avoid using Stopwatch Ticks.    Moreover we want the timestamps in Activity to be accurate.   Thus we should Make a method in Activity called 'GetUtcNow', which on CoreCLR just calls DateTime.UtcNow.    On Desktop it calls a version that uses Stopwatch (like what we did in our prototypes) to get a high precision version.   

Note that we should try to synchronize the two with code like

 sleep(1);   // This makes you synchronize the 16 msec tick clock.   
 long syncStopwatchTicks = Stopwatch.GetTimeStamp();
DateTime syncUtcNow = DateTime.UtcNow();

For extra credit, it keeps track of when the returned value is > 2 hour from the last sync time, and if so, spawns a Task to run the resynchronizing code (that keeps drift low, and fixes time updates)